[Swift in WebKit] Work towards modularizing WebCore private headers (Part 7)

rr-codes opened the pull request:
#### d964b116d6a6219406aa4d93723d36403c306140
<pre>
[Swift in WebKit] Work towards modularizing WebCore private headers (Part 7)
<a href="https://bugs.webkit.org/show_bug.cgi?id=297884">https://bugs.webkit.org/show_bug.cgi?id=297884</a>
<a href="https://rdar.apple.com/159146366">rdar://159146366</a>

Reviewed by Abrar Rahman Protyasha.

Fix some more module verification violations by adding missing headers and making private headers
project and vice versa.

* Source/WebCore/Scripts/SettingsTemplates/Settings.h.erb:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/scripts/CodeGeneratorJS.pm:
(IsPrivateHeader):
(GenerateHeader):
(WriteData):
* Source/WebCore/dom/make-event-names.py:
(main):
* Source/WebCore/dom/make_names.pl:
(printTypeHelpersHeaderFile):
(printNamesHeaderFile):
(printNodeNameHeaderFile):

Fix violations in generated JS binding header files by adjusting the relevant scripts to produce corrected output.

* Source/WebCore/html/canvas/WebGLExtensionAny.h:

Add missing includes.

Canonical link: <a href="https://commits.webkit.org/299174@main">https://commits.webkit.org/299174@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/74f6eba9fa6b5e108a5dc240e59e708ecce34771

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118127 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37805 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28441 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124282 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70166 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/37f7bb27-89aa-43d2-9c17-2f205cd0e6c7) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/120005 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38498 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46387 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/89642 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/59274 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/9ea33c9d-d9fd-4f1c-99f2-1a815c2fe7df) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121079 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30662 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105916 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70135 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/cb6d55e9-7d49-4a4f-af2f-3c373657b868) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29731 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/24033 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67951 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100088 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24210 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127359 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/45030 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/33938 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/98324 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45391 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102138 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98110 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24949 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43503 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21493 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/41475 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44902 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50576 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44362 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47707 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/46051 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebCore/bindings/scripts/CodeGeneratorJS.pm:
```diff
@@ -2121,6 +2121,65 @@ sub IsAlwaysExposedOnInterface
     return 1;
 }
 
+sub IsPrivateHeader
+{
+    my $headerFileName = shift;
+
+    # The set of DerivedSources headers that have private visibility.
+    # All private headers can only include other private headers or public headers.
+    # All private and public headers must be referenced using angle bracket includes.
+    # Always prefer using project visibility (instead of private) whenever possible.
+    my %privateHeaders = (
```
This is unlikely to change frequently, but is there a way to collect this list by calling into xcodebuild somehow? (rather than maintaining it statically)

rr-codes replied:
there is not, since (a) the script doesn't "know" about the Xcode build system, and (b) other platforms do not use Xcode. And ya it is very unlikely to change, and if it does it will clearly say what is wrong anyways.

pxlcoder replied:
Can we sort it at least?

rr-codes replied:
yes that I can do! (I started it sorted but then kinda lost lol will fix)

aprotyas replied:
> the script doesn't "know" about the Xcode build system

True, but also, this concept of private header teaches the script about the xcodebuild system... I wonder if we should move this to `CodeGeneratorJSExtensions+xcode.pm`, 

`use <config> qw(@IsPrivateHeader)`?

aprotyas replied:
up to you though, just thinking how to keep it neat

aprotyas replied:
(you'd probably want to name the abstraction something else and not `IsPrivateHeader`)

justinmichaud replied:
I would like to see this list moved into another file. For those of us not using xcode, this seems like a landmine.

Also, can this be read from the module map (which will be cross-platform when we adopt swift maybe)?

rr-codes replied:
> For those of us not using xcode, this seems like a landmine

This is not Xcode/Cocoa specific; it applies to all platforms.

rr-codes committed:
[Swift in WebKit] Work towards modularizing WebCore private headers (Part 7)
https://bugs.webkit.org/show_bug.cgi?id=297884
rdar://159146366

Reviewed by Abrar Rahman Protyasha.

Fix some more module verification violations by adding missing headers and making private headers
project and vice versa.

* Source/WebCore/Scripts/SettingsTemplates/Settings.h.erb:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/scripts/CodeGeneratorJS.pm:
(IsPrivateHeader):
(GenerateHeader):
(WriteData):
* Source/WebCore/dom/make-event-names.py:
(main):
* Source/WebCore/dom/make_names.pl:
(printTypeHelpersHeaderFile):
(printNamesHeaderFile):
(printNodeNameHeaderFile):

Fix violations in generated JS binding header files by adjusting the relevant scripts to produce corrected output.

* Source/WebCore/html/canvas/WebGLExtensionAny.h:

Add missing includes.

Canonical link: https://commits.webkit.org/299174@main
